### PR TITLE
user management docs

### PR DESCRIPTION
**Pull request by nilminiwso2**

### CODE REVIEW

File: en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md [hunk 1]
```diff
@@ -379,32 +386,67 @@ mi version
...
+       show             Get information about users
+
+    Flags:
+       -h, --help       Help for user
+       -p, --pattern    Specify a regex as a user name pattern
+       -r, --role       Specify a role to fetch a users        
+
+    Examples:
+       # To add a new user. This option is only available for admin users, therefore, set the `is-admin` flag to `true`
+       mi user add [new user-id] [password] [is-admin]
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md:407]: This is changed with https://github.com/wso2/product-mi-tooling/pull/31

File: en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md [hunk 2]
```diff
@@ -379,32 +386,67 @@ mi version
...
+       mi user show
+
+       # To list user by user ID
+       mi user show [user ID]
+
+       # To list users by user role
+       mi user show -r [role name]
+
+       # To list users matching a user name regex pattern
+       mi user show -p [user name regex pattern]
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md:422]: this is not exactly a regex. Just a filtering with * allowed at the start and the end.

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md [hunk 3]
```diff
@@ -74,10 +74,130 @@ Invoke the `/logout` resource to revoke the JWT token you used for [invoking the
...
-## Accessig API resources
+## Accessing API resources
 
 The management API has multiple resources to provide information regarding the deployed artifacts as well as the server itself.
 
+### GET USERS
+
+-	**Resource**: `/users`
+
+	**Description**: Retrieves a list of all users stored in an [external user store](../../../setup/user_stores/setting_up_ro_ldap).
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:85]: setting_up_**ro**_ldap. Is this correct?

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md [hunk 4]
```diff
@@ -74,10 +74,130 @@ Invoke the `/logout` resource to revoke the JWT token you used for [invoking the
...
+
+	**Example**:
+
+  	```bash tab='Request'
+  	curl -X GET "https://localhost:9164/management/users?pattern=”*us*”&role=”sole”" -H "accept: application/json" -H "Authorization: Bearer %AccessToken%" -k -i
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:90]: sole=role?

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md [hunk 5]
```diff
@@ -74,10 +74,130 @@ Invoke the `/logout` resource to revoke the JWT token you used for [invoking the
...
+  	curl -X GET "https://localhost:9164/management/users/user1" -H "accept: application/json" -H "Authorization: Bearer %AccessToken%" -k -i
+  	```
+
+    ```bash tab='Response'
+    {
+      userid: “user1”,
+      	isAdmin: true/false,
+      	roles :
+      [
+           role1,
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:121]: There's a small indentation issue. in line 117 as well.

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md [hunk 6]
```diff
@@ -74,10 +74,130 @@ Invoke the `/logout` resource to revoke the JWT token you used for [invoking the
...
+    ```
+
+-	**Resource**: `/users/pattern=”*”&role=admin`
+
+	**Description**: Retrieves information related to user names (stored in an [external user store](../../../setup/user_stores/setting_up_ro_ldap)) that match a specific regex pattern.
+
+	**Example**:
+
+  	```bash tab='Request'
+  	curl -X GET "https://localhost:9164/management/users?pattern=”*us*”&role=”sole”" -H "accept: application/json" -H "Authorization: Bearer %AccessToken%" -k -i
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:134]: sole -> role

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md [hunk 7]
```diff
@@ -74,10 +74,130 @@ Invoke the `/logout` resource to revoke the JWT token you used for [invoking the
...
+    	list:
+    	[
+    		userId: user1,
+    		userId: user2,
+    		userId: user3,
+    	]
+    }
+    ```
+
+### POST USERS
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:149]: Let's say add users.

File: en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md [hunk 8]
```diff
@@ -11,6 +11,25 @@ For instructions on installing DB2 Express-C, see this [ebook](https://www.ibm.c
 
 ## Setting up the database and users
 
+The following IBM DB scripts are stored in the `<MI_HOME>/dbscripts/` directory of your Micro Integrator.
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md:14]: IBM?

File: en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md [hunk 9]
```diff
@@ -65,4 +84,16 @@ pool_options.maxWait = 60000
 pool_options.testOnBorrow = true
 ```
 
+```toml tab='User DB Connection'
+[[datasource]]
+id = "WSO2_USER_DB"
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md:89]: We need to configure the realm to use this datasource
```
[realm_manager]
data_source = "WSO2_USER_DB"
```

Comment (sdkottegoda) [reply en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md:89]: same with all the db types.

File: en/micro-integrator/docs/setup/user_stores/setting_up_ro_ldap.md [hunk 10]
```diff
@@ -84,27 +78,87 @@ Follow the steps given below to connect the Micro Integrator to the LDAP user st
...
+	[[datasource]]
+	id = "WSO2_USER_DB"
+	url="jdbc:db2://SERVER_NAME:PORT/userdb"
+	username="root"
+	password="root"
+	driver="com.ibm.db2.jcc.DB2Driver"
+	pool_options.maxActive=50
+	pool_options.maxWait = 60000
+	pool_options.testOnBorrow = true
 	```
```
Comment (sdkottegoda) [inline en/micro-integrator/docs/setup/user_stores/setting_up_ro_ldap.md:158]: Are user store configurations missing? I couldn't locate.
```
[user_store]
class = "org.wso2.micro.integrator.security.user.core.jdbc.JDBCUserStoreManager"
type = "database"
read_only = false
```

Also we need to disable the file based user store for internal apis.
[internal_apis.file_user_store]
enable = false

Comment (nilminiwso2) [reply en/micro-integrator/docs/setup/user_stores/setting_up_ro_ldap.md:158]: @sdkottegoda 
The [user_store] is given under point 4 below.
Isn't the [internal_apis.file_user_store] disabled when we enable an external user store? Do we need to purposely disable the file-based user store?

File: en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md (hunk 9 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/setup/deployment/databases/setting-up-IBM-DB2.md:89]: fixed

File: en/micro-integrator/docs/setup/user_stores/setting_up_ro_ldap.md (hunk 10 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/setup/user_stores/setting_up_ro_ldap.md]: fixed.

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md (hunk 7 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/administer-and-observe/working-with-management-api.md]: Changed as follows:
- Add Users
- Activate/Deactivate Endpoints
- Activate/Deactivate Proxy Services
- Activate/Deactivate Message Processors

File: en/micro-integrator/docs/administer-and-observe/working-with-management-api.md (hunk 5 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/administer-and-observe/working-with-management-api.md:121]: fixed

File: en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md (hunk 2 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md]: fixed

File: en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md (hunk 1 above)
Comment (nilminiwso2) [reply en/micro-integrator/docs/administer-and-observe/using-the-command-line-interface.md:407]: Shall we apply these changes after merging this PR?

**Pull request merged**